sympy/concrete/summations.py

```diff
@@ -253,6 +257,202 @@ def _eval_simplify(self, ratio, measure):
+        try:
+            if limit(sequence_term, sym, upper_limit) != S.Zero:
+                return False
+        except NotImplementedError:
+            pass
+
+        order = O(sequence_term, (sym, S.Infinity))
+
+        ### --------- p-series test (1/n**p) ---------- ###
+        p1_series_test = order.expr.match(sym**p)
+        if p1_series_test is not None:
+            if p1_series_test[p] < -1:
+                return True
+            if p1_series_test[p] > -1:
+                return False
+
+        p2_series_test = order.expr.match((1/sym)**p)
+        if p2_series_test is not None:
+            if p2_series_test[p] > 1:
+                return True
+            if p2_series_test[p] < 1:
+                return False
+
+        ### ----------- root test ---------------- ###
+        lim = Limit(abs(sequence_term)**(1/sym), sym, S.Infinity)
+        lim_evaluated = lim.doit()
+        if lim_evaluated.is_number:
+            if lim_evaluated < 1:
+                return True
+            if lim_evaluated > 1:
+                return False
+
+        ### ------------- alternating series test ----------- ###
+        d = symbols('d', cls=Dummy)
+        dict_val = sequence_term.match((-1)**p*q)
+        if dict_val[p] == sym or dict_val[p].has(sym):
+            if is_decreasing(dict_val[q], interval):
+                return True
+            return False
+
+        ### ------------- comparison test ------------- ###
+        # (1/log(n)**p) comparison
+        log_test = order.expr.match(1/(log(sym)**p))
+        if log_test is not None:
+            return False
+
+        # (1/(n*log(n)**p)) comparison
+        log_n_test = order.expr.match(1/(sym*(log(sym))**p))
+        if log_n_test is not None:
+            if log_n_test[p] > 1:
+                return True
+            return False
+
+        # (1/(n*log(n)*log(log(n))*p)) comparison
+        log_log_n_test = order.expr.match(1/(sym*(log(sym)*log(log(sym))**p)))
+        if log_log_n_test is not None:
+            if log_log_n_test[p] > 1:
+                return True
+            return False
+
+        if order.expr == sequence_term:
+            pass
+        else:
+            return Sum(order.expr, (sym, lower_limit, upper_limit)).is_convergent()
+
+        ### ------------- integral test -------------- ###
+        if is_decreasing(sequence_term, interval):
+            integral_val = Integral(sequence_term, (sym, lower_limit, upper_limit))
+            try:
+                integral_val_evaluated = integral_val.doit()
+                if integral_val_evaluated.is_infinite:
+                    return False
+            except NotImplementedError:
+                pass
+            return True
+
+        ### -------------- Dirichlet tests -------------- ###
+        if order.expr.is_Mul:
+            a_n, b_n = order.expr.args[0], order.expr.args[1]
+
+            def _dirichlet_test(f_n, g_n):
+                if is_decreasing(f_n, interval):
```

Comment:
Though since `is_decreasing` is a part of `Dirichlet test`, so i thought it
to be better to keep it inside the function itself. None the less i have
changed as you said. Let me know what you think.

On Mon, Oct 5, 2015 at 3:05 PM, Kalevi Suominen <notifications@github.com>
wrote:

> In sympy/concrete/summations.py
> <https://github.com/sympy/sympy/pull/9906#discussion_r41125572>:
>
> > +        if is_decreasing(sequence_term, interval):
> > +            integral_val = Integral(sequence_term, (sym, lower_limit, upper_limit))
> > +            try:
> > +                integral_val_evaluated = integral_val.doit()
> > +                if integral_val_evaluated.is_infinite:
> > +                    return False
> > +            except NotImplementedError:
> > +                pass
> > +            return True
> > +
> > +        ### -------------- Dirichlet tests -------------- ###
> > +        if order.expr.is_Mul:
> > +            a_n, b_n = order.expr.args[0], order.expr.args[1]
> > +
> > +            def _dirichlet_test(f_n, g_n):
> > +                if is_decreasing(f_n, interval):
>
> This test could be taken before calling _dirichlet_test.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sympy/sympy/pull/9906/files#r41125572>.
>